File: branches/Medusa/Source/Fantasy.Jobs/Solar/SatelliteJobQueue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.Jobs.Management;
using Fantasy.ServiceModel;

namespace Fantasy.Jobs.Solar
{
    public class SatelliteJobQueue : AbstractService, IJobQueue
    {

        public override void InitializeService()
        {
            this._actionQueue = this.Site.GetRequiredService<ISolarActionQueue>();
            this._satellite = this.Site.GetRequiredService<ISatellite>();
            this._satellite.JobAdded += new EventHandler<JobQueueEventArgs>(SatelliteJobAdded);
            this._satellite.JobChanged += new EventHandler<JobQueueEventArgs>(SatelliteJobChanged);
            base.InitializeService();
        }

        void SatelliteJobChanged(object sender, JobQueueEventArgs e)
        {
            this.OnChanged(e);
        }

        void SatelliteJobAdded(object sender, JobQueueEventArgs e)
        {
            this.OnAdded(e);
        }


        private ISolarActionQueue _actionQueue;
        private ISatellite _satellite;

        #region IJobQueue Members

        public IEnumerable<JobMetaData> Unterminates
        {
            get 
            {
                using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
                {
                    int count = Int32.MaxValue;
                    int i = 0;
                    while (count >= 100)
                    {
                        JobMetaData[] rs = client.Client.Unterminates(i * 100, 100);
                        i += rs.Length;
                        count = rs.Length;

                        foreach (JobMetaData job in rs)
                        {
                            yield return job;
                        }
                    }
                }
                
            }
        }

        public IEnumerable<JobMetaData> Terminates
        {
            get
            {
                using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
                {
                    int count = Int32.MaxValue;
                    int i = 0;
                    while (count >= 100)
                    {
                        JobMetaData[] rs = client.Client.Terminates(i * 100, 100);
                        i += rs.Length;
                        count = rs.Length;

                        foreach (JobMetaData job in rs)
                        {
                            yield return job;
                        }
                    }
                }
            }
        }

        public JobMetaData FindJobMetaDataById(Guid id)
        {
            using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
            {
                return client.Client.FindJobMetaDataById(id);
            }
        }

        public IEnumerable<JobMetaData> FindTerminated(out int totalCount, string filter, string[] args, string order, int skip, int take)
        {
            using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
            {
                return client.Client.FindTerminated(out totalCount, filter, args, order, skip, take);
            }
        }

        public IEnumerable<JobMetaData> FindUnterminated(out int totalCount, string filter, string[] args, string order, int skip, int take)
        {
            using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
            {
                return client.Client.FindUnterminated(out totalCount, filter, args, order, skip, take);
            }
        }

        public IEnumerable<JobMetaData> FindAll()
        {
            return this.Unterminates.Union(this.Terminates);
        }

        public JobMetaData CreateJobMetaData()
        {
            return new JobMetaData();
        }

        public void ApplyChange(JobMetaData job)
        {
            _actionQueue.Enqueue(solar => solar.ApplyChange(job));
        }

        public void Resume(Guid id)
        {
            _actionQueue.Enqueue(solar => solar.Resume(id));
        }

        public void Cancel(Guid id)
        {
            _actionQueue.Enqueue(solar => solar.Cancel(id));
        }

        public void Suspend(Guid id)
        {
            _actionQueue.Enqueue(solar => solar.Suspend(id));
        }

        public void UserPause(Guid id)
        {
            _actionQueue.Enqueue(solar => solar.UserPause(id));
        }

        

        public event EventHandler<JobQueueEventArgs> Changed;

        protected virtual void OnChanged(JobQueueEventArgs e)
        {
            if (this.Changed != null)
            {
                this.Changed(this, e);
            }
        }


        

        public event EventHandler<JobQueueEventArgs> Added;

        protected virtual void OnAdded(JobQueueEventArgs e)
        {
            if (this.Added != null)
            {
                this.Added(this, e);
            }
        }


        event EventHandler<JobQueueEventArgs> IJobQueue.RequestCancel
        {
            add { throw new NotImplementedException(); }
            remove { throw new NotImplementedException(); }
        }

        event EventHandler<JobQueueEventArgs> IJobQueue.RequestSuspend
        {
            add { throw new NotImplementedException(); }
            remove { throw new NotImplementedException(); }
        }

        event EventHandler<JobQueueEventArgs> IJobQueue.RequestUserPause
        {
            add { throw new NotImplementedException(); }
            remove { throw new NotImplementedException(); }
        }

        public string[] GetAllApplications()
        {
            using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
            {
                return client.Client.GetAllApplications();
            }
        }

        public string[] GetAllUsers()
        {
            using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
            {
                return client.Client.GetAllUsers();
            }
        }

        public bool IsTerminated(Guid id)
        {
            using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
            {
                return client.Client.IsTermianted(id);
            }
        }

        #endregion
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/solar/SatelliteJobQueue.java
package fantasy.jobs.solar;

import Fantasy.Jobs.Management.*;
import Fantasy.ServiceModel.*;

public class SatelliteJobQueue extends AbstractService implements IJobQueue
{

	@Override
	public void InitializeService()
	{
		this._actionQueue = this.Site.<ISolarActionQueue>GetRequiredService();
		this._satellite = this.Site.<ISatellite>GetRequiredService();
//C# TO JAVA CONVERTER TODO TASK: Java has no equivalent to C#-style event wireups:
		this._satellite.JobAdded += new EventHandler<JobQueueEventArgs>(SatelliteJobAdded);
//C# TO JAVA CONVERTER TODO TASK: Java has no equivalent to C#-style event wireups:
		this._satellite.JobChanged += new EventHandler<JobQueueEventArgs>(SatelliteJobChanged);
		super.InitializeService();
	}

	private void SatelliteJobChanged(Object sender, JobQueueEventArgs e)
	{
		this.OnChanged(e);
	}

	private void SatelliteJobAdded(Object sender, JobQueueEventArgs e)
	{
		this.OnAdded(e);
	}


	private ISolarActionQueue _actionQueue;
	private ISatellite _satellite;

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IJobQueue Members

	public final Iterable<JobMetaData> getUnterminates()
	{
//C# TO JAVA CONVERTER NOTE: The following 'using' block is replaced by its Java equivalent:
//		using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
		ClientRef<ISolar> client = ClientFactory.<ISolar>Create();
		try
		{
			int count = Integer.MAX_VALUE;
			int i = 0;
			while (count >= 100)
			{
				JobMetaData[] rs = client.Client.Unterminates(i * 100, 100);
				i += rs.length;
				count = rs.length;

				for (JobMetaData job : rs)
				{
//C# TO JAVA CONVERTER TODO TASK: Java does not have an equivalent to the C# 'yield' keyword:
					yield return job;
				}
			}
		}
		finally
		{
		}

	}

	public final Iterable<JobMetaData> getTerminates()
	{
//C# TO JAVA CONVERTER NOTE: The following 'using' block is replaced by its Java equivalent:
//		using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
		ClientRef<ISolar> client = ClientFactory.<ISolar>Create();
		try
		{
			int count = Integer.MAX_VALUE;
			int i = 0;
			while (count >= 100)
			{
				JobMetaData[] rs = client.Client.Terminates(i * 100, 100);
				i += rs.length;
				count = rs.length;

				for (JobMetaData job : rs)
				{
//C# TO JAVA CONVERTER TODO TASK: Java does not have an equivalent to the C# 'yield' keyword:
					yield return job;
				}
			}
		}
		finally
		{
		}
	}

	public final JobMetaData FindJobMetaDataById(UUID id)
	{
//C# TO JAVA CONVERTER NOTE: The following 'using' block is replaced by its Java equivalent:
//		using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
		ClientRef<ISolar> client = ClientFactory.<ISolar>Create();
		try
		{
			return client.Client.FindJobMetaDataById(id);
		}
		finally
		{
		}
	}

	public final Iterable<JobMetaData> FindTerminated(RefObject<Integer> totalCount, String filter, String[] args, String order, int skip, int take)
	{
//C# TO JAVA CONVERTER NOTE: The following 'using' block is replaced by its Java equivalent:
//		using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
		ClientRef<ISolar> client = ClientFactory.<ISolar>Create();
		try
		{
			return client.Client.FindTerminated(totalCount, filter, args, order, skip, take);
		}
		finally
		{
		}
	}

	public final Iterable<JobMetaData> FindUnterminated(RefObject<Integer> totalCount, String filter, String[] args, String order, int skip, int take)
	{
//C# TO JAVA CONVERTER NOTE: The following 'using' block is replaced by its Java equivalent:
//		using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
		ClientRef<ISolar> client = ClientFactory.<ISolar>Create();
		try
		{
			return client.Client.FindUnterminated(totalCount, filter, args, order, skip, take);
		}
		finally
		{
		}
	}

	public final Iterable<JobMetaData> FindAll()
	{
		return this.getUnterminates().Union(this.getTerminates());
	}

	public final JobMetaData CreateJobMetaData()
	{
		return new JobMetaData();
	}

	public final void ApplyChange(JobMetaData job)
	{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		_actionQueue.Enqueue(solar => solar.ApplyChange(job));
	}

	public final void Resume(UUID id)
	{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		_actionQueue.Enqueue(solar => solar.Resume(id));
	}

	public final void Cancel(UUID id)
	{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		_actionQueue.Enqueue(solar => solar.Cancel(id));
	}

	public final void Suspend(UUID id)
	{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		_actionQueue.Enqueue(solar => solar.Suspend(id));
	}

	public final void UserPause(UUID id)
	{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		_actionQueue.Enqueue(solar => solar.UserPause(id));
	}



//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	public event EventHandler<JobQueueEventArgs> Changed;

	protected void OnChanged(JobQueueEventArgs e)
	{
		if (this.Changed != null)
		{
			this.Changed(this, e);
		}
	}




//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	public event EventHandler<JobQueueEventArgs> Added;

	protected void OnAdded(JobQueueEventArgs e)
	{
		if (this.Added != null)
		{
			this.Added(this, e);
		}
	}


//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	event EventHandler<JobQueueEventArgs> IJobQueue.RequestCancel
//		{
//			add
//			{
//				throw new NotImplementedException();
//			}
//			remove
//			{
//				throw new NotImplementedException();
//			}
//		}

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	event EventHandler<JobQueueEventArgs> IJobQueue.RequestSuspend
//		{
//			add
//			{
//				throw new NotImplementedException();
//			}
//			remove
//			{
//				throw new NotImplementedException();
//			}
//		}

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	event EventHandler<JobQueueEventArgs> IJobQueue.RequestUserPause
//		{
//			add
//			{
//				throw new NotImplementedException();
//			}
//			remove
//			{
//				throw new NotImplementedException();
//			}
//		}

	public final String[] GetAllApplications()
	{
//C# TO JAVA CONVERTER NOTE: The following 'using' block is replaced by its Java equivalent:
//		using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
		ClientRef<ISolar> client = ClientFactory.<ISolar>Create();
		try
		{
			return client.Client.GetAllApplications();
		}
		finally
		{
		}
	}

	public final String[] GetAllUsers()
	{
//C# TO JAVA CONVERTER NOTE: The following 'using' block is replaced by its Java equivalent:
//		using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
		ClientRef<ISolar> client = ClientFactory.<ISolar>Create();
		try
		{
			return client.Client.GetAllUsers();
		}
		finally
		{
		}
	}

	public final boolean IsTerminated(UUID id)
	{
//C# TO JAVA CONVERTER NOTE: The following 'using' block is replaced by its Java equivalent:
//		using (ClientRef<ISolar> client = ClientFactory.Create<ISolar>())
		ClientRef<ISolar> client = ClientFactory.<ISolar>Create();
		try
		{
			return client.Client.IsTermianted(id);
		}
		finally
		{
		}
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}